File: atinject-core/src/main/java/org/atinject/core/websocket/dto/WebSocketRequest.java
package org.atinject.core.websocket.dto;

import java.util.UUID;

import org.atinject.core.dto.DTO;

public abstract class WebSocketRequest extends DTO {
    
    private static final long serialVersionUID = 1L;

    private UUID requestId;
    private UUID rendezvous;
    
    public WebSocketRequest(){
        super();
    }

    public UUID getRequestId()
    {
        return requestId;
    }

    public WebSocketRequest setRequestId(UUID requestId)
    {
        this.requestId = requestId;
        return this;
    }

    public UUID getRendezvous() {
        return rendezvous;
    }

    public WebSocketRequest setRendezvous(UUID rendezvous) {
        this.rendezvous = rendezvous;
        return this;
    }
    
}

File: atinject-integration/src/main/resources/webapp/scripts/atinject/WebSocketRequest.js
define(["atinject/DTO"],
function(DTO){
    function WebSocketRequest() {
        this._class = "org.atinject.core.websocket.dto.WebSocketRequest"
    };

    WebSocketRequest.prototype = new DTO();

        WebSocketRequest.prototype.getRequestId = function() {
        return this.requestId;
    };

    WebSocketRequest.prototype.setRequestId = function(requestId) {
        this.requestId = requestId;
        return this;
    };

    WebSocketRequest.prototype.getRendezvous = function() {
        return this.rendezvous;
    };

    WebSocketRequest.prototype.setRendezvous = function(rendezvous) {
        this.rendezvous = rendezvous;
        return this;
    };


    
    return WebSocketRequest;
});
